### PR TITLE
Fix #3894 : Added bottomsheet

### LINKED ISSUE #3894: [A11Y] Unable to exit overflow menu
When a11y user is inside the overflow menu than currently there is no easy way to exit it. This might be a problem with the screen-reader (Talkback) itself as I have tested this on other apps like Whatsapp, Instagram and the experience is equally bad.

https://user-images.githubusercontent.com/9396084/136244259-9c79bc6a-ca2a-477f-9ec4-fec05447e99a.mp4

### PR DESCRIPTION
**Pull request by bkaur-bkj**
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by bkaur-bkj** [comment]
@rt4914 I created the bottom sheet fragment and tried implementing listeners on help and options button but I can't figure out a way to call the original function from within the fragment. I have also tried to paste the block of code in the fragment itself but it requires some dagger-related classes. PTAL

**Comment by rt4914** [comment]
> @rt4914 I created the bottom sheet fragment and tried implementing listeners on help and options button but I can't figure out a way to call the original function from within the fragment. I have also tried to paste the block of code in the fragment itself but it requires some dagger-related classes. PTAL

Have a look at https://github.com/oppia/oppia-android/blob/3a0afb4eb8222d4ff3acd2e2e660b85911d35229/app/src/main/java/org/oppia/android/app/player/stopplaying/StopExplorationDialogFragment.kt#L35 where we attach an interface to activity and similar to this link call the function of that interface. You can do something similar to communicate from BottomSheet to activity/fragment.

@bkaur-bkj I have tried the accessibility part and I have shared this video which shows how we don't require the "Close" item.

https://user-images.githubusercontent.com/9396084/150186405-7d453fb9-a68e-490f-83cc-7b5224cf7248.mp4

**Comment by bkaur-bkj** [comment]
> 

sir, I tried accessibility with bottom sheet after bottomsheet gets open if I swipe up my bottomsheet gets closed directly I don't get any such actions option. I see its fine but don't know why it shows different way

**Comment by rt4914** [comment]
> > 
> 
> sir, I tried accessibility with bottom sheet after bottomsheet gets open if I swipe up my bottomsheet gets closed directly I don't get any such actions option. I see its fine but don't know why it shows different way

@bkaur-bkj Can you please record and share the video?

**Comment by bkaur-bkj** [comment]
@rt4914  I am facing few dependency errors which I am unable to resolve - 
`symbol:   class BottomSheetItemClickListener
  location: package org.oppia.android.app.player.explorationC:\Users\Asus\Desktop\git_practice\oppia-android\app\build\tmp\kapt3\stubs\debug\org\oppia\android\app\player\exploration\ExplorationActivity.java:9: error: [ComponentProcessor:MiscError] dagger.internal.codegen.ComponentProcessor was unable to process this class because not all of its dependencies could be resolved. Check for compilation errors or a circular dependency with generated code.
    `

**Comment by bkaur-bkj** [comment]
Once this issue is resolved, I will mark the PR ready for review.

**Comment by bkaur-bkj** [comment]
> > > 
> > 
> > 
> > sir, I tried accessibility with bottom sheet after bottomsheet gets open if I swipe up my bottomsheet gets closed directly I don't get any such actions option. I see its fine but don't know why it shows different way
> 
> @bkaur-bkj Can you please record and share the video?

https://drive.google.com/file/d/16pGyh9XBpd7eCYU7iz6IvuQ1YIZkPzIN/view?usp=sharing this is the link of video

File: app/src/main/java/org/oppia/android/app/fragment/BottomSheetFragment.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  ): View? {
+    val binding = FragmentBottomSheetBinding.inflate(inflater,container,false)
+    setUpOnClickListeners(binding)
+    return binding.root
+  }
+
+  private fun setUpOnClickListeners(binding: FragmentBottomSheetBinding){
+    val bottomSheetItemClickListener = activity as BottomSheetItemClickListener
+    binding.actionHelp.setOnClickListener {
+      bottomSheetItemClickListener.handleOnOptionsItemSelected(R.id.action_help)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/fragment/BottomSheetFragment.kt]: I think once the click is captured, you should dismiss the dialog fragment. Same applies for other click too. This should mostly solve your error.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bkaur-bkj Thanks for the video. Its a bit unexpected but anyways we will be add close option to sheet.

File: app/src/main/java/org/oppia/android/app/fragment/BottomSheetFragment.kt (hunk 1 above)
Comment (bkaur-bkj) [reply app/src/main/java/org/oppia/android/app/fragment/BottomSheetFragment.kt]: @rt4914 I tried calling `dismiss()` both before and after calling `handleOnOptionsItemSelected` but it still gives the same error. I feel the error is not of dismiss because the app does not even compile.

**Comment by rt4914** [comment]
@bkaur-bkj I just checked out into your branch and ran the app and its working without any error. Can you try re-building once again? If the error still continues then we can meet and discuss it.

**Comment by bkaur-bkj** [comment]
> @bkaur-bkj I just checked out into your branch and ran the app and its working without any error. Can you try re-building once again? If the error still continues then we can meet and discuss it.

yes I tried rebuild, thanks it worked

**Comment by bkaur-bkj** [comment]
@rt4914 the functionality is working fine now, let me know the changes and should I add close option then? as u refused before but you mentioned in the last review that we will be adding it

**Comment by bkaur-bkj** [comment]
@rt4914 I have solved the other lint issues except imports to be lexicographically sorted,  I am unable to do that as my ktlint setup has some issues, can you pls help.

**Comment by bkaur-bkj** [comment]
@rt4914 also I am not able to understand about bazel test failure

**Comment by BenHenning** [comment]
@bkaur-bkj the problem is that whatever third party library is including BottomSheetFragment isn't part of the deps list for //app (https://github.com/oppia/oppia-android/blob/e20716bd2fc7580c0d1ab06e821c6373d1057a2a/app/BUILD.bazel#L719). You'll need to update that list to include the correct ``//third_party:<name>`` dependency so that the bottom sheet is available to use.

It may help to follow wiki instructions to set up Bazel locally so that you can build the app to verify that you've added the correct dependency and that it builds before you run CI again.

**Comment by bkaur-bkj** [comment]
> @bkaur-bkj the problem is that whatever third party library is including BottomSheetFragment isn't part of the deps list for //app (
> 
> https://github.com/oppia/oppia-android/blob/e20716bd2fc7580c0d1ab06e821c6373d1057a2a/app/BUILD.bazel#L719
> 
> ). You'll need to update that list to include the correct `//third_party:<name>` dependency so that the bottom sheet is available to use.
> It may help to follow wiki instructions to set up Bazel locally so that you can build the app to verify that you've added the correct dependency and that it builds before you run CI again.

@BenHenning I am not sure where to place this BottomSheetFragment.kt because currently it is inside `app/fragment` and this folder already has its BUILD.bazel. Also, I can't find over the internet import dependency for the bottom sheet, so can you help me with that also, please.

**Comment by rt4914** [comment]
@FareesHussain Can you please have a look https://github.com/oppia/oppia-android/pull/4099#issuecomment-1026519565

File: app/src/main/java/org/oppia/android/app/fragment/BottomSheetFragment.kt (hunk 1 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/android/app/fragment/BottomSheetFragment.kt]: Resolving this conversation as the error is solved

**Comment by bkaur-bkj** [comment]
@rt4914 have added the close button as we discussed

**Comment by FareesHussain** [comment]
@bkaur-bkj 

I think you need to add BottomSheetFragment to FragmentComponentImpl
and BottomSheetItemClickListener to [LISTENERS](https://github.com/oppia/oppia-android/blob/develop/app/BUILD.bazel#L85) 

and maybe add BottomSheetFragment.kt to [export_files](https://github.com/bkaur-bkj/oppia-android/blob/overflow-to-bottomsheet/app/src/main/java/org/oppia/android/app/fragment/BUILD.bazel#L8) if the build failes after doing the above changes. 


Let me know if any of above doesn't work

**Comment by bkaur-bkj** [comment]
> @bkaur-bkj
> 
> I think you need to add BottomSheetFragment to FragmentComponentImpl and BottomSheetItemClickListener to [LISTENERS](https://github.com/oppia/oppia-android/blob/develop/app/BUILD.bazel#L85)
> 
> and maybe add BottomSheetFragment.kt to [export_files](https://github.com/bkaur-bkj/oppia-android/blob/overflow-to-bottomsheet/app/src/main/java/org/oppia/android/app/fragment/BUILD.bazel#L8) if the build failes after doing the above changes.
> 
> Let me know if any of above doesn't work

@FareesHussain I tried your suggested changes but it still fails. PTAL

**Comment by BenHenning** [comment]
> > @bkaur-bkj the problem is that whatever third party library is including BottomSheetFragment isn't part of the deps list for //app (
> > https://github.com/oppia/oppia-android/blob/e20716bd2fc7580c0d1ab06e821c6373d1057a2a/app/BUILD.bazel#L719
> > 
> > ). You'll need to update that list to include the correct `//third_party:<name>` dependency so that the bottom sheet is available to use.
> > It may help to follow wiki instructions to set up Bazel locally so that you can build the app to verify that you've added the correct dependency and that it builds before you run CI again.
> 
> @BenHenning I am not sure where to place this BottomSheetFragment.kt because currently it is inside `app/fragment` and this folder already has its BUILD.bazel. Also, I can't find over the internet import dependency for the bottom sheet, so can you help me with that also, please.

@bkaur-bkj the issue is that you're introducing a dependency in Kotlin on com.google.android.material.bottomsheet.BottomSheetDialogFragment but the Bazel library that contains this class isn't part of the ``deps`` list for any of the libraries that are including references to that class (which leads to the unresolved symbol error). You need to add the third_party library that provides BottomSheetDialogFragment to the deps list of the library that includes files that reference it. I suggest looking at other third party dependencies to see how this is hooked up in both Bazel and Kotlin (such as ConstraintLayout).

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt [hunk 2]
```diff
@@ -466,970 +466,776 @@ class ExplorationActivityTest {
...
       )
-      openActionBarOverflowOrOptionsMenu(context)
-      onView(withText(context.getString(R.string.menu_options))).check(matches(isDisplayed()))
-      onView(withText(context.getString(R.string.menu_help))).check(matches(isDisplayed()))
+      onView(withId(R.id.action_bottom_sheet)).perform(click())
+      onView(withId(R.id.bottom_sheet_layout)).check(matches(isDisplayed()));
     }
-    explorationDataController.stopPlayingExploration()
   }
+    @Test
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt]: Add blank line above this.

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt [hunk 3]
```diff
@@ -466,970 +466,776 @@ class ExplorationActivityTest {
...
+          explorationCheckpoint = ExplorationCheckpoint.getDefaultInstance()
+        )
+        onView(withId(R.id.action_bottom_sheet)).perform(click())
+        onView(withText(context.getString(R.string.menu_help))).perform(click())
+        intended(hasComponent(HelpActivity::class.java.name))
+        intended(
+          hasExtra(
+            HelpActivity.BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY, /* value= */
+            false
+          )
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt]: Nit
```
          hasExtra(
            HelpActivity.BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY,
            /* value= */ false
          )
```

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bkaur-bkj The tests generally looks good. @BenHenning will be taking the lead on implementation part (as he is the code-owner) and I will be more focused towards the UI and accessibility parts.


In PR Description, make sure you have following things and also keep them up-to-date with your code always.
- Before screenshot/implementation video so that anyone can understand what was the implementation before.
- Reason for the changes - Why do we needs this big change?
- After screenshot / implementation video - mobile portrait/landscape, tablet portrait/landscape
- Talkback video and output showing that A11Y scanner passes on this new changes.

In general implementation is good. Thanks a lot.

**Comment by FareesHussain** [comment]
> @bkaur-bkj the issue is that you're introducing a dependency in Kotlin on com.google.android.material.bottomsheet.BottomSheetDialogFragment but the Bazel library that contains this class isn't part of the `deps` list for any of the libraries that are including references to that class (which leads to the unresolved symbol error). You need to add the third_party library that provides BottomSheetDialogFragment to the deps list of the library that includes files that reference it. I suggest looking at other third party dependencies to see how this is hooked up in both Bazel and Kotlin (such as ConstraintLayout).

Hi @BenHenning, I think you are missing something here. There wasn't any new dependency added. The BottomSheetFragment is a new file added to the fragment package. I suggested her to shift it to the same package as the ExplorationActivity to avoid the error.

Now I have a doubt here. I'm assuming that earlier BottomSheetFragment was not accessible to the player package, since the fragment package has its own BUILD.bazel which was the reason why this was not included in the glob of app/BUILD.bazel. Now what if I want to keep the BottomSheetFragment under the fragment package and I want it to be included within the glob (I think `APP_FILES_WITH_RESOURCE_IMPORTS`) from the app module without creating a separate target for BottomSheetFragment

**Comment by BenHenning** [comment]
Will follow up on this tomorrow--sorry for the delay.

**Comment by BenHenning** [comment]
> > @bkaur-bkj the issue is that you're introducing a dependency in Kotlin on com.google.android.material.bottomsheet.BottomSheetDialogFragment but the Bazel library that contains this class isn't part of the `deps` list for any of the libraries that are including references to that class (which leads to the unresolved symbol error). You need to add the third_party library that provides BottomSheetDialogFragment to the deps list of the library that includes files that reference it. I suggest looking at other third party dependencies to see how this is hooked up in both Bazel and Kotlin (such as ConstraintLayout).
> 
> Hi @BenHenning, I think you are missing something here. There wasn't any new dependency added. The BottomSheetFragment is a new file added to the fragment package. I suggested her to shift it to the same package as the ExplorationActivity to avoid the error.
> 
> Now I have a doubt here. I'm assuming that earlier BottomSheetFragment was not accessible to the player package, since the fragment package has its own BUILD.bazel which was the reason why this was not included in the glob of app/BUILD.bazel. Now what if I want to keep the BottomSheetFragment under the fragment package and I want it to be included within the glob (I think `APP_FILES_WITH_RESOURCE_IMPORTS`) from the app module without creating a separate target for BottomSheetFragment

@FareesHussain Ah, I did miss that--thanks for clarifying. The best thing to do here probably is to just add the library and not include it as part of the blob (we should be minimizing how much we add to those large libraries since we need to split them up, anyway). The new package is probably a better fit, anyway, since the new fragment is specific to the exploration player.

File: app/src/main/java/org/oppia/android/app/fragment/BUILD.bazel [hunk 4]
```diff
@@ -6,6 +6,7 @@ load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")
 
 # TODO(#59): Define these exported files as separate libraries from top-level targets.
 exports_files([
+    "BottomSheetFragment.kt",
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/fragment/BUILD.bazel]: This won't be needed anymore now that the fragment has been moved to player/exploration.

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetFragment.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package org.oppia.android.app.player.exploration
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetFragment.kt]: Is this specific to explorations, or is it also used by the question player? I think it might make more sense to put this under a shared package like player/options.

Further, we should also define a separate BUILD.bazel to build this fragment rather than including it in the top-level app module.

/cc @FareesHussain

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetFragment.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import org.oppia.android.R
+import org.oppia.android.databinding.FragmentBottomSheetBinding
+
+class BottomSheetFragment(val internalProfileId: Int) : BottomSheetDialogFragment() {
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetFragment.kt]: Values can't be passed into fragment constructors, they need to be passed in via arguments (which should use a proto--see other fragments that use protos for passing arguments or intent extras).

Regarding the name, 'BottomSheetFragment' is really generic. Could we name it more specifically to what it's going to be used for?

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -4,6 +4,7 @@
   <string name="menu_options">Options</string>
   <string name="menu_my_downloads">My Downloads</string>
   <string name="menu_help">Help</string>
+  <string name="menu_close">Close</string>
```
Comment (BenHenning) [inline app/src/main/res/values/strings.xml]: Prefer naming this more specifically to its context, e.g.: state_player_options_menu_close.

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt [hunk 8]
```diff
@@ -466,970 +466,776 @@ class ExplorationActivityTest {
...
-    launch<ExplorationActivity>(
-      createExplorationActivityIntent(
-        internalProfileId, RATIOS_TOPIC_ID,
-        RATIOS_STORY_ID_0, RATIOS_EXPLORATION_ID_0,
-        shouldSavePartialProgress = false
-      )
-    ).use {
-      explorationDataController.startPlayingExploration(
+    @Test
+    fun testAudioCellular_ratioExp_audioIcon_configChange_opensCellularAudioDialog() {
```
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt]: Here & elsewhere: what changed in all of these tests? I'm having a difficult time interpreting the deltas.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @bkaur-bkj. Took another pass--there are still failing CI checks that will need to be fixed (which will address some areas I didn't leave comments on). Once CI is passing I'll take a full review on the PR.

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt (hunk 2 above)
Comment (bkaur-bkj) [reply app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt (hunk 3 above)
Comment (bkaur-bkj) [reply app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt]: Done

**Comment by bkaur-bkj** [comment]
> @bkaur-bkj The tests generally looks good. @BenHenning will be taking the lead on implementation part (as he is the code-owner) and I will be more focused towards the UI and accessibility parts.
> 
> In PR Description, make sure you have following things and also keep them up-to-date with your code always.
> 
> * Before screenshot/implementation video so that anyone can understand what was the implementation before.
> * Reason for the changes - Why do we needs this big change?
> * After screenshot / implementation video - mobile portrait/landscape, tablet portrait/landscape
> * Talkback video and output showing that A11Y scanner passes on this new changes.
> 
> In general implementation is good. Thanks a lot.

Yes will be updating the description after completing the changes suggested

File: app/src/main/java/org/oppia/android/app/fragment/BUILD.bazel (hunk 4 above)
Comment (bkaur-bkj) [reply app/src/main/java/org/oppia/android/app/fragment/BUILD.bazel]: right, removed

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetFragment.kt (hunk 5 above)
Comment (bkaur-bkj) [reply app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetFragment.kt]: @FareesHussain Can you help me check on this

File: app/src/main/res/values/strings.xml (hunk 7 above)
Comment (bkaur-bkj) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetFragment.kt (hunk 6 above)
Comment (bkaur-bkj) [reply app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetFragment.kt]: Done, PTAL

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt (hunk 8 above)
Comment (bkaur-bkj) [reply app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt]: @BenHenning I have just added  tests , 1st for checking of Opening help activity, 2nd for Options activity and 3rd for closing of bottomsheet, rest in the audio tests I have not made any changes I am myself confused why git is showing that as a change in my commits, even though the content is exact same.

**Comment by bkaur-bkj** [comment]
@rt4914 I am not sure why this test fails for opening OptionsActivity from bottom sheet whereas the similar test for opening HelpActivity passes normally. PTAL

```
org.oppia.android.app.player.exploration.ExplorationActivityTest > testExploration_openBottomsheet_selectOptionsInBottomsheet_opensOptionsActivity FAILED
    androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: with text: is "Options"
```

**Comment by rt4914** [comment]
> testExploration_openBottomsheet_selectOptionsInBottomsheet_opensOptionsActivity

@bkaur-bkj Is your code updated to get this error? I checked out in your code and ran this test locally and it passed - twice.
Screenshot 2022-03-01 at 10 49 15 PM

For this I used Pixel 3A API 28

**Comment by bkaur-bkj** [comment]
> > testExploration_openBottomsheet_selectOptionsInBottomsheet_opensOptionsActivity
> 
> @bkaur-bkj Is your code updated to get this error? I checked out in your code and ran this test locally and it passed - twice. Screenshot 2022-03-01 at 10 49 15 PM
> 
> For this I used Pixel 3A API 28

But why are the same checks failing on Github Actions. PTAL

**Comment by rt4914** [comment]
@bkaur-bkj For now I have re-ran the CI checks as we have noticed something similar on #4183 also.

**Comment by bkaur-bkj** [comment]
> @bkaur-bkj For now I have re-ran the CI checks as we have noticed something similar on #4183 also.

Okay sir but I am not clear then how I should proceed as the checks failed again.

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenu.kt [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import org.oppia.android.R
+import org.oppia.android.databinding.FragmentBottomSheetBinding
+
+/** Bottom sheet fragment for displaying options menu */
+class BottomSheetOptionsMenu(val internalProfileId: Int) : BottomSheetDialogFragment() {
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenu.kt:12]: Does this parameter need to be passed in through arguments, instead, since Android can recreate the fragment itself?

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenu.kt [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  ): View? {
+    val binding = FragmentBottomSheetBinding.inflate(inflater, container, false)
+    setUpOnClickListeners(binding)
+    return binding.root
+  }
+
+  private fun setUpOnClickListeners(binding: FragmentBottomSheetBinding) {
+    val bottomSheetItemClickListener = activity as BottomSheetOptionsMenuItemClickListener
+    binding.actionHelp.setOnClickListener {
+
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenu.kt:27]: Nit: please remove extra newline for consistency. Ditto below.

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * Listener for executing the correct function based on the options clicked in the
+ * [BottomSheetOptionsMenu].
+ */
+interface BottomSheetOptionsMenuItemClickListener {
+  /**
+   * Function for passing the selected itemId
+   *
+   * @param itemId: Resource Id of the option selected
+   * @return [Unit]
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt:12]: This isn't needed for ``Unit`` return types.

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * Listener for executing the correct function based on the options clicked in the
+ * [BottomSheetOptionsMenu].
+ */
+interface BottomSheetOptionsMenuItemClickListener {
+  /**
+   * Function for passing the selected itemId
+   *
+   * @param itemId: Resource Id of the option selected
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt:11]:
```suggestion
   * @param itemId resource Id of the option selected
```

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+   * @return [Unit]
+   */
+  fun handleOnOptionsItemSelected(itemId: Int)
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt:14]: Perhaps use an enum instead of a resource ID to limit the possibilities of what can be passed to this function (limiting possibilities generally results in more error-tolerant and understandable code).

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt [hunk 14]
```diff
@@ -0,0 +1,15 @@
+package org.oppia.android.app.player.exploration
+
+/**
+ * Listener for executing the correct function based on the options clicked in the
+ * [BottomSheetOptionsMenu].
+ */
+interface BottomSheetOptionsMenuItemClickListener {
+  /**
+   * Function for passing the selected itemId
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt:9]: Perhaps instead: ``Called when the user selects a customization option mid-play session.``

File: app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt [hunk 15]
```diff
@@ -0,0 +1,15 @@
+package org.oppia.android.app.player.exploration
+
+/**
+ * Listener for executing the correct function based on the options clicked in the
+ * [BottomSheetOptionsMenu].
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/exploration/BottomSheetOptionsMenuItemClickListener.kt:5]: Perhaps instead: ``Listener for when the user interacts with the play session options bottom sheet.``

File: app/src/main/res/layout/fragment_bottom_sheet.xml [hunk 16]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (BenHenning) [inline app/src/main/res/layout/fragment_bottom_sheet.xml:1]: The name of this file should match its location since all layout resources are global to the app.

File: app/src/main/res/layout/exploration_activity.xml [hunk 17]
```diff
@@ -55,6 +55,14 @@
             android:scaleType="center"
             app:srcCompat="@{viewModel.isAudioStreamingOn ? @drawable/ic_audio_streaming_on_24dp : @drawable/ic_audio_streaming_off_24dp}"
             android:visibility="@{viewModel.showAudioButton ? View.VISIBLE : View.GONE}" />
+
+          <ImageButton
+            android:id="@+id/action_bottom_sheet"
```
Comment (BenHenning) [inline app/src/main/res/layout/exploration_activity.xml:60]: IDs are global to the app--this should be more specific to what specific menu it corresponds to.

Ditto for the options in the bottom sheet fragment layout below.

File: app/src/main/res/layout/fragment_bottom_sheet.xml [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+<layout>
+  <androidx.appcompat.widget.LinearLayoutCompat xmlns:android="http://schemas.android.com/apk/res/android"
+  android:id="@+id/bottom_sheet_layout"
+  android:layout_width="match_parent"
+  android:layout_height="wrap_content"
+  android:padding="10dp"
+  android:orientation="vertical">
```
Comment (BenHenning) [inline app/src/main/res/layout/fragment_bottom_sheet.xml:8]: Indentation seems off here--please fix.

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt [hunk 19]
```diff
@@ -444,7 +443,7 @@ class ExplorationActivityTest {
   }
 
   @Test
-  fun testExploration_overflowMenu_isDisplayedSuccessfully() {
+  fun testExploration_openBottomSheet() {
```
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt:450]: What's the expected outcome? The action seems to be opening the bottom sheet, but it's not clear what this test is making sure works.

Perhaps instead:

``testActivity_initialState_openBottomSheet_showsBottomSheet``

Might help clarify?

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt [hunk 20]
```diff
@@ -1839,7 +1867,8 @@ class ExplorationActivityTest {
 
   private fun addShadowMediaPlayerException(dataSource: Any, exception: Exception) {
     val classLoader = ExplorationActivityTest::class.java.classLoader!!
-    val shadowMediaPlayerClass = classLoader.loadClass("org.robolectric.shadows.ShadowMediaPlayer")
+    val shadowMediaPlayerClass =
+      classLoader.loadClass("org.robolectric.shadows.ShadowMediaPlayer")
```
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt:1875]: Why this change? Ditto above. I suspect these might be extra and unnecessary reformats done by Android Studio.

**Review by BenHenning — COMMENTED** [review]
Thanks @bkaur-bkj. Took a pass--PTAL.

I'm also going to dig into why the CI checks aren't running; I'm going to see if I can get them to restart.

**Comment by BenHenning** [comment]
It looks like CI checks are running now. It may help also to merge the latest develop in @bkaur-bkj as it contains improvements for Gradle runs.
